packages/transformer/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -399,39 +405,40 @@ describe("IModelTransformer", () => {
         "==============================="
       );
       const targetImporter = new RecordingIModelImporter(targetDb);
-      const transformer = new TestIModelTransformer(sourceDb, targetImporter);
+      const transformer = new TestIModelTransformer(sourceDb, targetImporter, {
+        forceExternalSourceAspectProvenance: true,
+      });
       await transformer.processAll();
       assert.equal(targetImporter.numModelsInserted, 0);
       assert.equal(targetImporter.numModelsUpdated, 0);
       assert.equal(targetImporter.numElementsInserted, 1);
       assert.equal(targetImporter.numElementsUpdated, 33);
-      // FIXME: upgrade this test to use a briefcase so that we can detect element deletes
-      // use the new force old detect deletes behavior flag here
-      // assert.equal(targetImporter.numElementsDeleted, 5);
+      /**
+       * There are 5 elements deleted in TransformerExtensiveTestScenario.updateDb, but only 4 detected.
+       * This is because PhysicalObject6's code is scoped to PhysicalObject5. When PhysicalObject5 is deleted, PhysicalObject6 is also deleted in the superclasses
+       * of the RecordingIModelImporter and therefore can't be detected by the RecordingIModelImporter.
```

Comment:
maybe this is a good time to stop using `RecordingIModelImporter`? You could alternatively change the tree deletions (iirc this is the problem) to return the set of deletes somehow for the RecordIModelImporter to process. But I'd rather we not perform an iModel insert for every delete... I think it's awkward and probably brittle.

However you do this, assert that the fifth element is no longer in the iModel, and maybe rename this to `numElementsExplicitlyDeleted` and document on that property that it doesn't include implicit deletions when deleting trees.

Reply:
I ended up keeping RecordingIModelImporter as is but, I did change the numElementsDeleted to explicitlyDeleted.

Regarding 'return the set of deletes somehow for the RecordIModelImporter to process', I did have a brief thought about this that any custom importers might run into this issue if they're doing something for each delete, but I'm hoping thats not the case and I'd rather defer this problem for the future / if theres a request or need for it.